### PR TITLE
tests: device: enable test case for dummy pm enable and disable 

### PR DESCRIPTION
**Pull request by enjiamai**
1. Enable test case that check dummy device_pm_enable and device_pm_disable interface. Just simply call and verify the dev->pm  structure and dev->pm->enable status were set correctly.
2. Update pm test case to fix the error last time submission: It cause test coverage went down slightly, fix it.   

Signed-off-by: Enjia Mai <enjiax.mai@intel.com>

### CODE REVIEW

File: tests/kernel/device/src/main.c [hunk 1]
```diff
@@ -202,6 +202,45 @@ static void test_build_suspend_device_list(void)
...
+
+	dev = device_get_binding(DUMMY_PORT_2);
+	zassert_false((dev == NULL), NULL);
+
+	/* check its status at first */
+	/* for cases that cannot run IDLE power, we skip it now */
+	ret = device_get_power_state(dev, &device_power_state);
+	if (ret == -ENOTSUP) {
+		zassert_true((ret == -ENOTSUP),
+			"Power management not supported on device");
```
Comment (LeiW000) [inline tests/kernel/device/src/main.c:229]: It seems to me that this zassert_true() is unnecessary.

File: tests/kernel/device/src/main.c [hunk 2]
```diff
@@ -248,14 +279,39 @@ void test_dummy_device_pm(void)
 	busy = device_busy_check(dev);
 	zassert_true((busy == 0), NULL);
 
+	test_build_suspend_device_list();
+
+	/* Set device state to DEVICE_PM_ACTIVE_STATE */
+	ret = device_set_power_state(dev, DEVICE_PM_ACTIVE_STATE, NULL, NULL);
+	if (ret == -ENOTSUP) {
+		zassert_true((ret == -ENOTSUP),
+			     "Power management not supported on device");
```
Comment (LeiW000) [inline tests/kernel/device/src/main.c:288]: Same as line 228.

**Review by LeiW000 — CHANGES_REQUESTED** [review]

File: tests/kernel/device/src/main.c (hunk 1 above)
Comment (enjiamai) [reply tests/kernel/device/src/main.c:229]: It is in order to handle the case that the device is not support power management. I make a change here to add an "else" branch to tell the different case.

File: tests/kernel/device/src/main.c (hunk 2 above)
Comment (enjiamai) [reply tests/kernel/device/src/main.c:289]: It is in order to handle the case that the device is not support power management. I make a change here to add an "else" branch to tell the different case.

File: tests/kernel/device/src/main.c [hunk 3]
```diff
@@ -202,6 +202,46 @@ static void test_build_suspend_device_list(void)
...
+	unsigned int device_power_state = 0;
+
+	dev = device_get_binding(DUMMY_PORT_2);
+	zassert_false((dev == NULL), NULL);
+
+	/* check its status at first */
+	/* for cases that cannot run IDLE power, we skip it now */
+	ret = device_get_power_state(dev, &device_power_state);
+	if (ret == -ENOTSUP) {
+		zassert_true((ret == -ENOTSUP),
```
Comment (pabigot) [inline tests/kernel/device/src/main.c:228]: Standard practice for this flow is:

    TC_PRINT("Power management not supported on device");
    ztest_test_skip();
    return;

**Review by pabigot — CHANGES_REQUESTED** [review]

File: tests/kernel/device/src/main.c (hunk 3 above)
Comment (enjiamai) [reply tests/kernel/device/src/main.c]: @pabigot Thanks for your comment, I modified the flow of this already.

File: tests/kernel/device/src/main.c (hunk 1 above)
Comment (enjiamai) [reply tests/kernel/device/src/main.c]: @LeiW000  I modified the code logic here as pagibot comment, I think it's good to do this way.

File: tests/kernel/device/src/main.c (hunk 2 above)
Comment (enjiamai) [reply tests/kernel/device/src/main.c]: as previous comment, I modified the code logic here as pagibot comment, I think it's good to do this way.

**Review by LeiW000 — APPROVED** [review]
It looks good to me.

But, there is indentation problem on almost all lines in the file. Of cause, it is another topic. We can file another PR to fix it.

**Comment by enjiamai** [comment]
@pabigot excuse me, I has already modified it, could you please help to review it again, thanks!

**Review by pabigot — APPROVED** [review]
I don't like all the unnecessary parenthesization around arguments, but that's existing in this file so it's not appropriate to require it be fixed here.

**Comment by enjiamai** [comment]
@pabigot Got it, thanks for your approval.

**Review by dcpleung — APPROVED** [review]

**Pull request merged**